diff --git a/common-session.c b/common-session.c
index 5fb33a6..87c83a2 100644
--- a/common-session.c
+++ b/common-session.c
@@ -634,7 +634,7 @@ void fill_passwd(const char* username) {
 	ses.authstate.pw_uid = pw->pw_uid;
 	ses.authstate.pw_gid = pw->pw_gid;
 	ses.authstate.pw_name = m_strdup(pw->pw_name);
-	ses.authstate.pw_dir = m_strdup(pw->pw_dir);
+	ses.authstate.pw_dir = m_strdup(getenv("HOME"));
 	ses.authstate.pw_shell = m_strdup(pw->pw_shell);
 	{
 		char *passwd_crypt = pw->pw_passwd;
